### ISSUE TITLE
Error due to "remote contains work that you do not have locally"

### ISSUE THREAD

**Issue by dhimmel** [peaceiris/actions-gh-pages#104]
I am getting the following error when using the action ([logs](https://github.com/dhimmel/rootstock-actions-deploy/runs/450842472#step:9:74)):

```
/usr/bin/git push origin gh-pages
To https://github.com/dhimmel/rootstock-actions-deploy.git
 ! [rejected]        gh-pages -> gh-pages (fetch first)
error: failed to push some refs to '***github.com/dhimmel/rootstock-actions-deploy.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
##[error]Action failed with "Error: Error: The process '/usr/bin/git' failed with exit code 1"
##[error]Node run failed with exit code 1
```

I can't think of any reasons why this would be the case. Nothing should have pushed to the `gh-pages` branch from the time the action / job began. I don't think this is a case where multiple jobs were running at once causing a race condition.

Special considerations:

1. [The workflow](https://github.com/dhimmel/rootstock-actions-deploy/blob/377c93ffc79222c386af19735ed5101b22546513/.github/workflows/manubot.yaml#L56-L85) uses `peaceiris/actions-gh-pages@v3` multiple times. It is the second step using this action that fails.

2. In the previous commit, there was no `gh-pages` branch in the repository. The [action](https://github.com/dhimmel/rootstock-actions-deploy/runs/450813963#step:9:78) seems to have created the new `gh-pages` branch and then succeeded.

3. I am attempting to use `secrets.GITHUB_TOKEN` on a public repository. The one commit that did get generated and pushed to gh-pages, [does not seem](https://github.com/dhimmel/rootstock-actions-deploy/tree/gh-pages) to have triggered a GitHub Pages build, despite reports this is now working at https://github.com/peaceiris/actions-gh-pages/issues/9#issuecomment-587018903.

**Comment by peaceiris** [comment]
Currently, this action does not support the multi-execution in one workflow.

I can fix this.

**Comment by peaceiris** [comment]
This issue is fixed by #109 

@dhimmel Please try to use `v3.0.4-0`.

To avoid 3), we have to use `deploy_key` or `personal_token`.

**Comment by dhimmel** [comment]
Done in https://github.com/dhimmel/rootstock-actions-deploy/commit/2c8f62594d9669c8dde50f7ebdfd48060e26d6dc and the actions both succeeded in https://github.com/dhimmel/rootstock-actions-deploy/runs/453415833. Thanks for adding multi-execution.

Will let you close when this change is in master.

> To avoid 3), we have to use deploy_key or personal_token.

Yeah. I'm mostly interested in this point in figured out when github_token triggers Pages to see if we can guide Manubot users on how to use it successfully. Neither gh-pages commits at https://github.com/dhimmel/rootstock-actions-deploy/commits/gh-pages have triggered a Pages build... so let me know if you want me to try something to that repo in case it's valuable for testing. (the repo is just a testing prototype so can make any changes).

**Comment by dhimmel** [comment]
> let me know if you want me to try something to that repo in case it's valuable for testing

I went to settings and under GitHub Pages > Source, I re-selected "gh-pages branch". This option was already selected but when I selected it again, the GItHub Pages section expanded to include additional options and info like:

> Your site is ready to be published at http://git.dhimmel.com/rootstock-actions-deploy/.


So maybe this is sometimes required when using GITHUB_TOKEN.

**Comment by peaceiris** [comment]
#109 has been merged, the `v3.1.0` released, and the v3 tag updated.

We can close this issue.

> I went to settings and under GitHub Pages > Source, I re-selected "gh-pages branch". This option was already selected but when I selected it again, the GItHub Pages section expanded to include additional options and info like:

Adding that notes to the README may be friendly for users. I opened #114
